tests/upgrades/test_client.py

```diff
@@ -27,373 +28,303 @@
+    default_org, pre_upgrade_cv, pre_upgrade_repo, module_ak
+):
+    """Create product and repo, from which a package will be installed
+    post upgrade. Create a content host and register it.
+
+    :id: preupgrade-eedab638-fdc9-41fa-bc81-75dd2790f7be
+
+    :setup:
+
+        1. Create and sync repo from which a package can be
+            installed on content host
+        2. Add repo to CV and then to Activation key
+
+
+    :steps:
+
+        1. Create a container as content host and register with Activation key
+
+    :expectedresults:
+
+        1. The "pre-upgrade" content host is created and registered.
+        2. The new repo is enabled on the content host.
+
+    """
+    rhel7_client = dockerize(ak_name=module_ak.name, distro='rhel7', org_label=default_org.label)
+    client_container_id = list(rhel7_client.values())[0]
+    # Wait 30 seconds as script in /tmp takes about 15s but only starts after 10s
+    time.sleep(30)
+    # Assert gofer was installed
+    installed_package = execute(
+        docker_execute_command,
+        client_container_id,
+        'rpm -q gofer',
+        host=docker_vm,
+    )[docker_vm]
+    assert 'package gofer is not installed' not in installed_package
+    # Run goferd on client as its docker container
+    kwargs = {'async': True, 'host': docker_vm}
+    execute(docker_execute_command, client_container_id, 'goferd -f', **kwargs)
+    # Holding on for 30 seconds while goferd starts
+    time.sleep(30)
+    status = execute(docker_execute_command, client_container_id, 'ps -aux', host=docker_vm)[
+        docker_vm
+    ]
+    assert 'goferd' in status
+    # Save client info to disk for post-upgrade test
+    create_dict({__name__: rhel7_client})
+
+
+@post_upgrade(depend_on=test_pre_scenario_preclient_package_installation)
+def test_post_scenario_preclient_package_installation(default_org):
+    """Post-upgrade install of a package on a client created and registered pre-upgrade.
+
+    :id: postupgrade-eedab638-fdc9-41fa-bc81-75dd2790f7be
+
+    :steps: Install package on the pre-upgrade registered client
+
+    :expectedresults: The package is installed on client
+    """
+    client = get_entity_data(__name__)
+    client_name = str(list(client.keys())[0]).lower()
+    client_id = entities.Host().search(query={'search': f'name={client_name}'})[0].id
+    entities.Host().install_content(
+        data={
+            'organization_id': default_org.id,
+            'included': {'ids': [client_id]},
+            'content_type': 'package',
+            'content': [FAKE_0_CUSTOM_PACKAGE],
+        }
+    )
+    # Verifies that package is really installed
+    installed_package = execute(
+        docker_execute_command,
+        list(client.values())[0],
+        f'rpm -q {FAKE_0_CUSTOM_PACKAGE}',
+        host=docker_vm,
+    )[docker_vm]
+    assert FAKE_0_CUSTOM_PACKAGE in installed_package
+
+    """This section contains post-upgrade scenarios to test if the package
```

Comment:
This also misses on class or deleted the existing one !

Reply:
I have now put them back.